Neos.Flow/Classes/ObjectManagement/Proxy/ProxyClass.php

```diff
@@ -302,4 +334,68 @@ protected function renderTraitsCode()
 
         return '    use ' . implode(', ', $this->traits) . ";\n\n";
     }
+
+    /**
+     * @param $proxyClassName
+     * @param string $originalClassName
+     * @return string
+     * @throws ReflectionException
+     */
+    protected function buildLazyProxyClass($proxyClassName, string $originalClassName): string
+    {
+        $methods = [$this->buildCallMagicMethod()];
+
+        foreach (get_class_methods($this->fullOriginalClassName) as $methodName) {
+            if (!$this->reflectionService->isMethodPublic($this->fullOriginalClassName, $methodName) || $methodName === '__call' || $methodName === '__construct') {
+                continue;
+            }
+            $methodReturnType = $this->reflectionService->getMethodDeclaredReturnType($this->fullOriginalClassName, $methodName);
+            $returnKeyword = ($methodReturnType === 'void') ? '' : 'return ';
+            $method = MethodGenerator::fromReflection(new \Laminas\Code\Reflection\MethodReflection($this->fullOriginalClassName, $methodName));
+            $method->removeDocBlock();
+            $method->setBody(
+                <<< CODE
+                    \$arguments = func_get_args();
+                    {$returnKeyword}\$this->_activateDependency()->{$methodName}(...\$arguments);
+                CODE
+            );
+            $methods[] = $method;
+        }
+
+        $classGenerator = ClassGenerator::fromArray(['name' => $proxyClassName . '_LazyProxy'])
+            ->setExtendedClass($proxyClassName)
+            ->setImplementedInterfaces([DependencyProxy::class])
+            ->removeMethod('__call')
+            ->addMethods($methods)
+            ->addTrait('\\' . DependencyProxyTrait::class);
+        // This is just for making Psalm happy, since addTrait() returns TraitUsageInterface which doesn't provide generate()
+        assert($classGenerator instanceof ClassGenerator);
+        return $classGenerator->generate();
+    }
+
+    /**
+     * @return MethodGenerator
+     * @throws ReflectionException
+     */
+    protected function buildCallMagicMethod(): MethodGenerator
+    {
+        if (method_exists($this->fullOriginalClassName, '__call')) {
+            $callMagicMethod = MethodGenerator::fromReflection(new \Laminas\Code\Reflection\MethodReflection($this->fullOriginalClassName, '__call'));
+        } else {
+            $callMagicMethod = MethodGenerator::fromArray(['name' => '__call']);
+            $callMagicMethod->setParameters([
+                ParameterGenerator::fromArray(['name' => 'methodName', 'type' => 'string']),
+                ParameterGenerator::fromArray(['name' => 'arguments', 'type' => 'array']),
+            ]);
+        }
+
+        $callMagicMethod->setBody(
+            <<< CODE
+                [\$methodName, \$arguments] = func_get_args();
+                return \$this->_activateDependency()->\$methodName(...\$arguments);
+            CODE
+        );
```

Comment:
isnt usign `func_get_args` always not so cool?
https://github.com/neos/flow-development-collection/pull/2701#discussion_r834056321

would be a minor performance boost in some cases: but should not block this pr!

```suggestion
        if (method_exists($this->fullOriginalClassName, '__call')) {
            $callMagicMethod = MethodGenerator::fromReflection(new \Laminas\Code\Reflection\MethodReflection($this->fullOriginalClassName, '__call'));
            $callMagicMethod->setBody(
                <<<'CODE'
                    [$methodName, $arguments] = func_get_args();
                    return $this->_activateDependency()->$methodName(...$arguments);
                CODE
             );
        } else {
            $callMagicMethod = MethodGenerator::fromArray(['name' => '__call']);
            $callMagicMethod->setParameters([
                ParameterGenerator::fromArray(['name' => 'methodName', 'type' => 'string']),
                ParameterGenerator::fromArray(['name' => 'arguments', 'type' => 'array']),
            ]);
            $callMagicMethod->setBody(
                <<<'CODE'
                    return $this->_activateDependency()->$methodName(...$arguments);
                CODE
            );
        }
```

Reply:
otherwise the intention about why we even care to do so: is not so clear to me ...

```
            $callMagicMethod->setParameters([
                ParameterGenerator::fromArray(['name' => 'methodName', 'type' => 'string']),
                ParameterGenerator::fromArray(['name' => 'arguments', 'type' => 'array']),
            ]); 
```

Reply:
thank you for your time and work already, and sorry for beeing maybe too hard on this pr - i just want to be a good reviewer. I will test this later locally and then martins 1/2 review and my half share are 1 so then we can maybe merge already ^^